src/js/helpers/Util.js

```diff
@@ -105,8 +105,8 @@ var Util = {
     } else if (this.isArray(obj)) {
       copy = obj.slice(); // shallow copy
 
-      copy.forEach((value, index) => {
-        copy[index] = this.deepCopy(value);
+      Object.keys(copy).forEach((key) => {
```

Comment:
I think this is the Holzweg. :)

We are testing on an array above, `.keys()` only applies on an object.